src/app/Services/ProductService.ts

```diff
@@ -0,0 +1,58 @@
+import { Injectable } from "@angular/core";
+import { ProductItem } from "src/ProductItem";
+import { Observable } from "rxjs";
+import { backendURL } from "src/utils";
+import { HttpClient } from "@angular/common/http";
+import { ShoppingCartItem } from "src/ShoppingCardItem";
+
+@Injectable({providedIn:'root'})
+export class ProductService{
+
+
+    constructor(
+        private httpClient:HttpClient
+    ){}
+    add:Boolean=false;
+    show:Boolean=false;
+    product:ProductItem | undefined;
+    items:ShoppingCartItem[]=[]
+
+
+
+    public addProduct(product:ProductItem):Observable<ProductItem>{
+        return this.httpClient.post<ProductItem>((backendURL+"products"),product)
+    }
+
+    public showForm(id:Number):void{
+        this.show=!this.show
+        this.httpClient.get(`${backendURL}products/${id}`).subscribe((data:any)=>{this.product=data})
+    }
+
+    public editProduct(product:ProductItem):Observable<ProductItem>{
+        return this.httpClient.put<ProductItem>((backendURL+'products/'+product.id),product)
+    }
+
+    public checkOut(item:ShoppingCartItem):void{
+        console.log(item)
```

Comment:
**suggestion:** you should not keep the console.log in your code after debugging